.github/workflows/rust.yaml

```diff
@@ -14,9 +14,21 @@ jobs:
     steps:
       - name: Checkout sources
         uses: actions/checkout@v2
-      - name: Add toolchain
-        run: rustup target add wasm32-unknown-unknown        
-      - name: Run checks
+      - name: Run all checks
         run: make check
+  test:
+    name: Tests
+    runs-on: [self-hosted, heavy]
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - name: Add toolchain target
```

Comment:
I guess this job is redundant on the self-hosted runner. The target `wasm32-unknown-unknown` is already present there.